src/metabase/driver/sql_jdbc/sync/describe_table.clj

```diff
@@ -414,27 +414,58 @@
                                         (false? (:json_unfolding existing-field))))]
         (remove should-not-unfold? json-fields)))))
 
+(defn- sample-json-row-honey-sql
+  "Return a hoenysql used to get row sample to describe json columns.
+
+  If the talbe has a PK, try to fetch both first and last rows (see ##25744).
+  Else fetch the first n rows only"
+  [driver table-identifier json-field-names pk-identifier]
+  (binding [hx/*honey-sql-version* (sql.qp/honey-sql-version driver)]
+    (let [pk-expr          (sql.qp/maybe-wrap-unaliased-expr pk-identifier)
+          table-expr       (sql.qp/maybe-wrap-unaliased-expr table-identifier)
+          json-field-names (mapv sql.qp/maybe-wrap-unaliased-expr json-field-names)]
+      (if pk-identifier
+        {:select json-field-names
+         :from   [table-expr]
+         ;; mysql doesn't support limit in where, so we're using inner join here
+         :join  [[{:union [{:nest {:select   [pk-expr]
```

Comment:
I'm not following, why don't we want the result to be `distinct`? it's my intention to have the queries only get distinct rows

Reply:
I'm saying either have the results be distinct in both cases, or not distinct in both cases. Here you have one case of the `(if pk-identifier ...)` block where the results are distinct with `union`, and the other there the results are not distinct. If it's your intention to have distinct results you should apply `distinct` to the second case to make them consistent.

Reply:
The json fields could have duplicates.

It would need to be 
```
{:select-distinct json-field-names
 :from   [table-expr]
 :limit  metadata-queries/nested-field-sample-limit}
```

Reply:
the sql version for case having a pk only do distinct on the pk. duplicated json rows would still be returned.

Also, it's not possible to select distinct a json column cause the equality operator is not supported for json type.

Reply:
You're right, my apologies